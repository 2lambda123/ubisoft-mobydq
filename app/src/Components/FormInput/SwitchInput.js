import React from 'react';
import PropTypes from 'prop-types';
import Switch from '@material-ui/core/Switch';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import InputFeedback from './InputFeedback';

const SwitchInput = ({
  id,
  label,
  error,
  touched,
  errors,
  value,
  onChange,
  className,
  ...props
}) => {
  return (
    <FormControlLabel
      control={
        <React.Fragment>
          <Switch
            id={id}
            onChange={onChange}
            errorText={touched && errors}
            value={value}
            {...props}
          />
          < InputFeedback error={error} />
        </React.Fragment>
      }
      label={label}
    />
  );
};

export default SwitchInput;


SwitchInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};