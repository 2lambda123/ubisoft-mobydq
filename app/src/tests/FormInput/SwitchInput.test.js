import React from 'react'
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SwitchInput from '../../Components/FormInput/SwitchInput';

describe('SwitchInput unit test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SwitchInput
        id="Id"
        label="Label"
        touched={jest.fn()}
        error={false}
        value="value"
        onChange={jest.fn()}
        onBlur={jest.fn()}
      />
    );
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});