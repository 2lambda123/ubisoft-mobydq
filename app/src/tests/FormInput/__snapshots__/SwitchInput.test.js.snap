// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchInput unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SwitchInput
    error={false}
    id="Id"
    label="Label"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    touched={[MockFunction]}
    value="value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(FormControlLabel)
          control={
            <React.Fragment>
              <WithStyles(Switch)
                id="Id"
                onBlur={[MockFunction]}
                onChange={[MockFunction]}
                value="value"
              />
            </React.Fragment>
          }
          label="Label"
        />,
        <WithStyles(FormHelperText) />,
      ],
      "error": false,
      "style": Object {
        "margin": "15px",
        "width": "250px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "control": <React.Fragment>
            <WithStyles(Switch)
              id="Id"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              value="value"
            />
          </React.Fragment>,
          "label": "Label",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": false,
        },
        "ref": null,
        "rendered": false,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(FormControlLabel)
            control={
              <React.Fragment>
                <WithStyles(Switch)
                  id="Id"
                  onBlur={[MockFunction]}
                  onChange={[MockFunction]}
                  value="value"
                />
              </React.Fragment>
            }
            label="Label"
          />,
          <WithStyles(FormHelperText) />,
        ],
        "error": false,
        "style": Object {
          "margin": "15px",
          "width": "250px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "control": <React.Fragment>
              <WithStyles(Switch)
                id="Id"
                onBlur={[MockFunction]}
                onChange={[MockFunction]}
                value="value"
              />
            </React.Fragment>,
            "label": "Label",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": false,
          },
          "ref": null,
          "rendered": false,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
